### DISCUSSION TITLE
Cache encodings for smooth playback on low-end hardware

### DISCUSSION THREAD

**Discussion by alfonsocv12** [jellyfin/jellyfin#11393] [General]
Hello, first of all thanks a lot for the great work, I love using Jellyfin at my house, my wire love it to.

I wanted to share this idea (Cache encodings) and request for comments on it.

As a POC I'm thinking on adding the support at `VideosController.cs` where if it finds the video with the format required by the client device, creates a streaming of that file instead of starting encoding the original video.

The tradeoff is storage, you offload GPU power with storage, but this is a great addition for hi-bitrate videos that are imposible to encode and stream smoothly with low-end hardware (I'm using a 15 inch macbook pro from 2011)

**Comment by gnattu** [comment]
This is an old feature request to create a pre-transcoded file:

https://features.jellyfin.org/posts/284/convert-option-similar-to-plexs-optimise

The problem of your specific Mac is, with Jellyfin 10.9 moving to dotnet 8, that will require macOS 12 to run, and your Mac is too old to be officially supported on that macOS version, so you may need to install an alternate OS or use some unofficial way to force macOS 12 to install.

**Comment by alfonsocv12** [comment]
Yes is really similar, 

The main difference is that I think the biggest bottleneck is the wait time of encoding and not the in transit speeds, I think for most low-end hardware is possible to send 4k files but its really hard to encode and have smooth playback.

I say this because in this feature suggestion they say the want to save a lower size encoding and be able to select that from the video player.

The idea of this Cache encodings is just to save the file encoded, really similar to what Netflix OpenConnect servers have, they don't put big GPUs in them, what they do is put big hard-drives and save all of the encodings, here is a [talk](https://www.youtube.com/watch?v=q4TZxj-Dq7s&ab_channel=OpenFestBulgaria) about the topic by Drew Gallatin a Netflix engineer.

For the Macbook support, I run Debian 12 on the laptop so I have a pretty modern OS.
